src/ui/display.rs

```
@@ -108,23 +110,30 @@ impl ScreenShooter {
         let (image, xres, yres) = {
             let fb = &self.inner.lock().unwrap().0;
 
-            let bpp = (fb.var_screen_info.bits_per_pixel / 8) as usize;
-            let xres = fb.var_screen_info.xres;
-            let yres = fb.var_screen_info.yres;
-            let res = (xres as usize) * (yres as usize);
+            assert!(fb.var_screen_info.bits_per_pixel == 32);
```

Instead of a potential panic please return an appropriate error in this case.